### ISSUE TITLE
Azure CLI browser login redirect fails with WSL >= 1.1.3

### ISSUE THREAD

**Issue by cmendible** [microsoft/WSL#9921]
### Windows Version

Microsoft Windows [Version 10.0.23424.1000]

### WSL Version

1.1.7.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

6.1.21.1-1

### Distro Version

Ubuntu 22.04

### Other Software

azure-cli                         2.47.0
Windows Terminal Version: 1.16.10262.0

### Repro Steps

Login to azure running:

``` bash
az login
```

once in the browser enter your credentials. The redirect to `localhost:port` will fail

---

Running `ss -lntp` inside WSL result:

``` bash
State Recv-Q Send-Q Local Address:Port Peer Address:Port Process
LISTEN 0 5 127.0.0.1:45049 0.0.0.0:* users:(("python3",pid=839,fd=4))
```

---

Running `netstat -an | findstr /c:"LISTENING"` on Windows:

``` bash
TCP 0.0.0.0:135 0.0.0.0:0 LISTENING
TCP 0.0.0.0:445 0.0.0.0:0 LISTENING
TCP 0.0.0.0:5040 0.0.0.0:0 LISTENING
TCP 0.0.0.0:49664 0.0.0.0:0 LISTENING
TCP 0.0.0.0:49665 0.0.0.0:0 LISTENING
TCP 0.0.0.0:49666 0.0.0.0:0 LISTENING
TCP 0.0.0.0:49667 0.0.0.0:0 LISTENING
TCP 0.0.0.0:49668 0.0.0.0:0 LISTENING
TCP 0.0.0.0:49669 0.0.0.0:0 LISTENING
TCP 127.0.0.1:2015 0.0.0.0:0 LISTENING
TCP 127.0.0.1:29678 0.0.0.0:0 LISTENING
TCP 172.21.96.1:139 0.0.0.0:0 LISTENING
TCP 192.168.1.104:139 0.0.0.0:0 LISTENING
TCP [::]:135 [::]:0 LISTENING
TCP [::]:445 [::]:0 LISTENING
TCP [::]:49664 [::]:0 LISTENING
TCP [::]:49665 [::]:0 LISTENING
TCP [::]:49666 [::]:0 LISTENING
TCP [::]:49667 [::]:0 LISTENING
TCP [::]:49668 [::]:0 LISTENING
TCP [::]:49669 [::]:0 LISTENING
```

### Expected Behavior

The redirect to `localhost:port`  completes and login process is successful.

### Actual Behavior

login process inside WSL keeps waiting until canceled by the user.

**Comment by benhillis** [comment]
Thanks @cmendible - I tried this locally and wasn't able to repro. Could you please collect /logs?

**Comment by pmartincic** [comment]
@cmendible, Please run `wsl --shutdown` before starting log collection, repro the issue, and then stop log collection.

**Comment by cmendible** [comment]
Issue also present in 1.2.0.0 👇

WSL version: 1.2.0.0
Kernel version: 5.15.90.1
WSLg version: 1.0.51
MSRDC version: 1.2.3770
Direct3D version: 1.608.2-61064218
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.23424.1000

The logs:

[WslLogs-2023-04-07_08-36-32.zip](https://github.com/microsoft/WSL/files/11176997/WslLogs-2023-04-07_08-36-32.zip)

**Comment by cmendible** [comment]
> Thanks @cmendible - I tried this locally and wasn't able to repro. Could you please collect /logs?

Are you running aarch64?

Logs shared 👆

**Comment by benhillis** [comment]
Thanks for the logs - that had exactly what I needed. Good news is the fix is quite straightforward, hope to push it out later this week.

**Comment by benhillis** [comment]
Fixed in https://github.com/microsoft/WSL/releases/tag/1.2.1.

**Comment by johananl** [comment]
Workaround which worked for me: Copy the URL from the browser and run `curl <url>` inside WSL.

**Comment by johananl** [comment]
> Fixed in https://github.com/microsoft/WSL/releases/tag/1.2.1.

I confirm this is fixed using 1.2.2 (ran `wsl --update --pre-release` to update WSL).
